### DISCUSSION TITLE
OpenSSL with zlib occur issues

### DISCUSSION THREAD

**Discussion by whoseas** [openssl/openssl#24654] [Q&A] [answered]
Below is my command

openssl 1.1.1d    zlib 1.2.11

perl Configure VC-WIN32 --prefix=D:\xxx\openssl --with-zlib-include=D:\xxx\zlib\include --with-zlib-lib=D:\xxx\zlib\lib\zlibstat.lib zlib

and I get below issues, is someone can help.

![image](https://github.com/openssl/openssl/assets/9656553/0038b23b-9c54-4a15-ad99-3fcafac6468c)

**Comment by nhorman** [comment]
openssl 1.1.1d is long out of support

Does the same issue occur with openssl 3.3?

**Reply by whoseas** [reply]
hi，yes i have tried with the latest OpenSSL + the latest zlib, still same issue.

**Comment by nhorman** [comment]
I just tried it with zlib here, and it works fine.  I did notice that you are linking with zlibstat.lib, which seemed odd, and is likely causing your errors if you pass:
```
--with-zlib-lib=/path/to/zlib/zlib.lib
```
it should work fine

**Reply by whoseas** [reply]
Hi Neil,

Thanks a lot for your reply, but the problem is zlibstat.lib is a static library of zlib (not an import library of zlib.dll), and my command is to use "zlib" instead of "zlib-dynamic".  so the purpose is to build openssl with zlib static library.
please correct me if my command is wrong.

perl Configure VC-WIN32 --prefix=D:\xxx\openssl --with-zlib-include=D:\xxx\zlib\include --with-zlib-lib=D:\xxx\zlib\lib\zlibstat.lib **_zlib_**

And according to the INSTALL.md, the LIB is the filename of zlib library, so I think use "zlibstat.lib" is also OK , and whether your zlib.lib is a static library? 
--with-zlib-lib=LIB
                   On Windows: this is the filename of the zlib library (with or
                   without a path). This flag must be provided if the
                   zlib-dynamic option is not also used. If zlib-dynamic is used
                   then this flag is optional and a default value ("ZLIB1") is
                   used if not provided. 

kindly advise.
thanks a lot.

**Reply by whoseas** [reply]
Another information, the zlibstat.lib is generated from xxx\zlib-1.3.1\contrib\vstudio\vc14  run .sln in visual studio 2017 and build zlibstat project.

**Comment by nhorman** [comment]
try adding no-zlib-dynamic to your openssl config.  It seems zlib-dynamic is implicitly set when you select zlib

**Reply by whoseas** [reply, marked as answer]
Hi Neil,
Unfortunatelly, no-zlib-dynamic is not working for me.
but I find that use Makefile.msc to generate the static zlib.lib is working to build the openssl.
so the solution for me is to use Makefile.msc instead of use the .sln to genterate the static zlib lib.

thanks a lot for your help/response.
